src/test/operators/table_scan_test.cpp

```diff
@@ -1072,4 +1114,155 @@ TEST_P(OperatorsTableScanTest, TwoBigScans) {
   }
 }
 
+/**
+ * Tests for sorted_by flag forwarding.
+ */
+TEST_P(OperatorsTableScanTest, KeepOrderByFlagUnset) {
+  // Verify that the sorted_by flag is not set when it's not present in left input.
+  const auto scan_unsorted =
+      create_table_scan(get_int_float_op(), ColumnID{0}, PredicateCondition::GreaterThanEquals, 1234);
+  scan_unsorted->execute();
+
+  const auto& result_table_unsorted = scan_unsorted->get_output();
+
+  EXPECT_TRUE(result_table_unsorted->get_chunk(ChunkID{0})->sorted_by().empty());
+  EXPECT_TRUE(result_table_unsorted->get_chunk(ChunkID{1})->sorted_by().empty());
+}
+
+TEST_P(OperatorsTableScanTest, SortedFlagDataSegments) {
+  // In the current implementation, scans on tables that do not contain ReferenceSegments are executed chunk-by-chunk.
+  // As each input chunks will lead to one output chunk (assuming there are qualifying tuples), the position is set to
+  // guarantee that only a single chunk is referenced. Hence, the scan maintains the sort order that was initially set
+  // and forwards the sorted_by flag.
+  const auto scan_sorted =
+      create_table_scan(get_int_sorted_op(), ColumnID{0}, PredicateCondition::LessThanEquals, 1234);
+  scan_sorted->execute();
+  const auto& result_chunk_sorted = scan_sorted->get_output()->get_chunk(ChunkID{0});
+
+  const auto ref_segment =
+      std::dynamic_pointer_cast<const ReferenceSegment>(result_chunk_sorted->get_segment(ColumnID{0}));
+  EXPECT_TRUE(ref_segment->pos_list()->references_single_chunk());
+
+  const auto expected_sorted_by =
+      std::vector<SortColumnDefinition>{SortColumnDefinition(ColumnID{0}, SortMode::Ascending)};
+  EXPECT_EQ(result_chunk_sorted->sorted_by(), expected_sorted_by);
+  EXPECT_EQ(result_chunk_sorted->sorted_by(), expected_sorted_by);
+}
+
+// Reference segments that only reference a single chunk allow forwarding the sorting flags.
+TEST_P(OperatorsTableScanTest, SortedFlagReferenceSegments) {
+  const auto table = get_int_sorted_op()->table;
+
+  std::shared_ptr<Table> ref_table = std::make_shared<Table>(table->column_definitions(), TableType::References);
+
+  auto pos_list_1 = std::make_shared<RowIDPosList>();
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 1});
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 0});
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 3});
+  pos_list_1->emplace_back(RowID{ChunkID{0}, 2});
+  pos_list_1->guarantee_single_chunk();
+
+  const auto segment_1 = std::make_shared<ReferenceSegment>(table, ColumnID{0}, pos_list_1);
+  ref_table->append_chunk({segment_1});
+
+  auto pos_list_2 = std::make_shared<RowIDPosList>();
+  pos_list_2->emplace_back(RowID{ChunkID{1}, 0});
+  pos_list_2->guarantee_single_chunk();
+
+  const auto segment_2 = std::make_shared<ReferenceSegment>(table, ColumnID{0}, pos_list_1);
+  ref_table->append_chunk({segment_2});
```

Comment:
You probably meant pos_list_2, but changing it makes the test fail.

Reply:
Fixed.